kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/source/reader/ResultReader.scala

```diff
@@ -41,17 +41,8 @@ class ResultReader(
     * Retrieves the results for a particular reader, or None if no further results are available
     */
   def retrieveResults(limit: Int): Option[Vector[SourceRecord]] = {
-
     val results: Vector[SourceRecord] = accumulate(limit, reader, Vector.empty[SourceRecord])
-
-    if (results.isEmpty) {
-      None
-    } else {
-      Some(
-        results,
-      )
-    }
-
+    Option.when(results.nonEmpty)(results)
```

Comment:
Just a remark that this could also work `results.headOption.map(_=> results)`.

Reply:
I think I prefer it the original way, taking the headOption and ignoring the value seems a little untidy